app/scripts/controllers/permissions/selectors.js

```diff
@@ -44,6 +44,33 @@ export const getPermittedAccountsByOrigin = createSelector(
   },
 );
 
+/**
+ * Get the authorized CAIP-25 scopes for each subject, keyed by origin.
+ * The values of the returned map are immutable values from the
+ * PermissionController state.
+ *
+ * @returns {Map<string, Caip25Authorization>} The current origin:authorization map.
+ */
+export const getAuthorizedScopesByOrigin = createSelector(
```

Comment:
Should these selectors be memoized for performance?

Reply:
`createSelector` memoizes by default: https://redux.js.org/usage/deriving-data-selectors#:~:text=Reselect%20provides%20a%20function%20called,function%20for%20you%20to%20use.

Reply:
Yes, but we generally use our own function that does deeper equality checks for the memoization. https://github.com/MetaMask/metamask-extension/blob/600e4b10adfd48faa79467b651eeced9d2765095/ui/selectors/selectors.js#L1110

Otherwise the UI may heavily re-render since the entire state may be replaced when the background syncs to the frontend

Reply:
```suggestion
export const getAuthorizedScopesByOrigin = createDeepEqualSelector(
```

Reply:
done here https://github.com/MetaMask/metamask-extension/pull/27782/commits/8a5f2cff5b8e32ab0a412ddbabba8b82fb6d6602